### DISCUSSION TITLE
HTTP Exception - MIRTH 4.4.1

### DISCUSSION THREAD

**Discussion by boukenka** [nextgenhealthcare/connect#5956] [Frequently Asked Questions] [answered]
Hello,

Via the template, there is a download of information from an API REST. Below is the code.

```
  var url = new java.net.URL('http://host.docker.internal:5000/.....');
  var conn = url.openConnection();
  conn.setConnectTimeout(40000);
  conn.setReadTimeout(60000);
  conn.setDoOutput(true);
  conn.setRequestMethod('GET');
  conn.setRequestProperty('Accept', 'application/json');
  conn.setRequestProperty('Content-Type', 'application/json');
  conn.setRequestProperty("Authorization", 'XXXX...');
```

I get the following error : 

_JavaException: java.lang.IllegalAccessException: class org.mozilla.javascript.MemberBox cannot access class [sun.net](http://sun.net/).[www.protocol.http.HttpURLConnection](http://www.protocol.http.httpurlconnection/) (in module java.base) because module java.base does not export [sun.net](http://sun.net/).[www.protocol.http](http://www.protocol.http/) to unnamed module @1eba372c_

In the MIRTH docker, via the terminal, a CURL doing the same call with the same parameters is working.
I am using MIRTH **4.4.1**.

Any help would be greatly appreciated.

**Comment by boukenka** [comment]
With the same code, it **works with MIRTH 4.0.0**.

**Reply by boukenka** [reply]
It works as well in **MIRTH 4.1.1.** It doesn't work in **MIRTH 4.2.0**.

**Comment by narupley** [comment, marked as answer]
This probably depends on your Java distribution and whether you have added this into your vmoptions:

```--add-exports=java.base/sun.net.www.protocol.http=ALL-UNNAMED```

**Reply by boukenka** [reply]
Thank you for the answer. But how to use this option ? I am launching the MIRTH Docker instance with the following command.

`docker run -v c:/Demo:/Demo -p 8444:8443 nextgenhealthcare/connect:4.2.0`

**Reply by pacmano1** [reply]
see https://hub.docker.com/r/nextgenhealthcare/connect/, where it discusses what env vars you can set.

**Reply by narupley** [reply]
Yeah you'd want to use that `VMOPTIONS` envvar for the Docker container.

**Reply by boukenka** [reply]
Thank you both for your answers ! It works :smile:

`docker run -e VMOPTIONS=--add-exports=java.base/sun.net.www.protocol.http=ALL-UNNAMED -v c:/Demo:/Demo -p 8444:8443 nextgenhealthcare/connect:4.2.0`
